cmd/admin-handlers.go

```diff
@@ -520,10 +523,15 @@ func (a adminAPIHandlers) HealHandler(w http.ResponseWriter, r *http.Request) {
 			case hr := <-respCh:
 				switch {
 				case hr.errCode == ErrNone:
-					writeSuccessResponseJSON(w, hr.respBytes)
-				case hr.errBody == "":
+					if started {
+						w.Write(hr.respBytes)
+						w.(http.Flusher).Flush()
+					} else {
+						writeSuccessResponseJSON(w, hr.respBytes)
+					}
+				case hr.errBody == "" && !started:
 					writeErrorResponseJSON(w, hr.errCode, r.URL)
-				default:
+				case !started:
```

Comment:
Looks like the case when `started && hr.errCode != ErrNone` is not handled in this switch case? We would need to write error message without response headers to the http writer.

Reply:
That would be incorrect - client code will become complex.  Also this is a one time operation the actual status per heal is written by a different code path.

Reply:
Actually you may be right, need to add an extra `if` to make it work. But need to change client code as well to handle it.

Reply:
Done